src/legend.ts

```diff
@@ -1,115 +1,88 @@
 import {DateTime} from './datetime';
-import {VgLegendEncode} from './vega.schema';
+import {VgLegendEncode, VgLegendBase} from './vega.schema';
 
-export interface LegendConfig {
-  // ---------- General ----------
-  /**
-   * Optional mark definitions for custom legend encoding.
-   */
-  encode?: VgLegendEncode;
-  /**
-   * Padding (in pixels) between legend entries in a symbol legend.
-   */
-  entryPadding?: number;
-  /**
-   * The orientation of the legend. One of "left" or "right". This determines how the legend is positioned within the scene. The default is "right".
-   */
-  orient?: string;
-  /**
-   * The offset, in pixels, by which to displace the legend from the edge of the enclosing group or data rectangle.
-   */
-  offset?: number;
-  /**
-   * The padding, in pixels, between the legend and axis.
-   */
-  padding?: number;
-  /**
-   * The margin around the legend, in pixels
-   */
-  margin?: number;
-  /**
-   * The number of ticks for legend.
-   */
-  tickCount?: number;
-  /**
-   * A non-positive integer indicating z-index of the legend.
-   * If zindex is 0, legend should be drawn behind all chart elements.
-   * To put them in front, use zindex = 1.
-   * @TJS-type integer
-   * @minimum 0
-   */
-  zindex?: number;
+export interface LegendConfig extends LegendBase {
   // ---------- Gradient ----------
   /**
    * The color of the gradient stroke, can be in hex color code or regular color name.
    */
   gradientStrokeColor?: string;
+
   /**
    * The width of the gradient stroke, in pixels.
    * @mimimum 0
    */
   gradientStrokeWidth?: number;
+
   /**
    * The height of the gradient, in pixels.
    * @mimimum 0
    */
   gradientHeight?: number;
+
   /**
    * The width of the gradient, in pixels.
    * @mimimum 0
    */
   gradientWidth?: number;
+
   // ---------- Label ----------
   /**
    * The alignment of the legend label, can be left, middle or right.
    */
   labelAlign?: string;
+
   /**
    * The position of the baseline of legend label, can be top, middle or bottom.
    */
   labelBaseline?: string;
+
   /**
    * The color of the legend label, can be in hex color code or regular color name.
    */
   labelColor?: string;
+
   /**
    * The font of the legend label.
    */
   labelFont?: string;
+
   /**
    * The font size of legend label.
    * @mimimum 0
    */
   labelFontSize?: number;
+
   /**
    * The offset of the legend label.
    * @mimimum 0
    */
   labelOffset?: number;
-  /**
-   * Whether month names and weekday names should be abbreviated.
-   */
-  shortTimeLabels?: boolean;
+
   // ---------- Symbols ----------
   /**
    * The color of the legend symbol,
    */
   symbolColor?: string;
+
   /**
    * The shape of the legend symbol, can be the 'circle', 'square', 'cross', 'diamond',
    * 'triangle-up', 'triangle-down', or else a custom SVG path string.
    */
-  symbolShape?: string;
+  symbolType?: string;
```

Comment:
`symbolShape` isn't renamed to `symbolType`.  `symbolType` is simply a new property.  Please

- [ ] correct description
- [ ] update the PR description / commit message  (rename => remove + add)